### PR TITLE
[openmvg, netcdf-c] Update ports

### PR DESCRIPTION
**Pull request by AlvinZhangH**
1. Update port netcdf-c version from 4.7.0-5 to 4.7.2, openmvg version from 1.4-6 to 1.5.
2. Fix patch.
3. No features of port netcdf-c need to test. Features opencv and openmp of port openmvg install passed on x86-windows.

### CODE REVIEW

File: ports/netcdf-c/hdf5_3.patch [hunk 1]
```diff
@@ -12,3 +12,4 @@ index f3c7bbc..34fc2ab 100644
 +else()
 + target_link_libraries(netcdfhdf5 PRIVATE hdf5::hdf5-static hdf5::hdf5_hl-static)
```
Comment (JackBoosY) [inline ports/netcdf-c/hdf5_3.patch]: This line should be discussed.
Related: https://github.com/microsoft/vcpkg/pull/9043#discussion_r348946218.

File: ports/openmvg/fixcmake.patch [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 +++ b/src/CMakeLists.txt
-@@ -17,27 +17,16 @@ endif()
- # ==============================================================================
- # OpenMVG build options
- # ==============================================================================
+@@ -14,31 +14,6 @@ if (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+   message(FATAL_ERROR "In-source builds not allowed.")
+ endif()
+ 
+-# ==============================================================================
```
Comment (JackBoosY) [inline ports/openmvg/fixcmake.patch]: Could you use `if (0) ... endif()` instead of delete them?

File: ports/openmvg/fixcmake.patch [hunk 3]
```diff
@@ -81,29 +85,33 @@ index 654bebd..ab7dd90 100644
  # ==============================================================================
  # --END-- SUBMODULE CONFIGURATION
  # ==============================================================================
-@@ -226,14 +192,8 @@ ENDMACRO(UNIT_TEST)
+@@ -225,18 +177,8 @@ ENDMACRO(UNIT_TEST)
  # - external by default
  # - internal if cereal not found
  # ==============================================================================
 -find_package(cereal QUIET CONFIG)
```
Comment (JackBoosY) [inline ports/openmvg/fixcmake.patch]: Could you use `if (0) ... endif()` instead of delete them?

File: ports/openmvg/fixcmake.patch [hunk 4]
```diff
@@ -81,29 +85,33 @@ index 654bebd..ab7dd90 100644
...
 +get_target_property(CEREAL_INCLUDE_DIRS cereal INTERFACE_INCLUDE_DIRECTORIES)
  
  # ==============================================================================
  # Eigen
-@@ -242,14 +202,8 @@ endif()
+@@ -245,14 +187,8 @@ endif()
  # - external if EIGEN_INCLUDE_DIR_HINTS is defined
  # - internal if Eigen not found
  # ==============================================================================
--find_package(eigen3 QUIET)
```
Comment (JackBoosY) [inline ports/openmvg/fixcmake.patch]: Could you use `if (0) ... endif()` instead of delete them?

File: ports/openmvg/fixcmake.patch [hunk 5]
```diff
@@ -150,60 +158,60 @@ index 654bebd..ab7dd90 100644
  
  # ==============================================================================
  # CoinUtils
-@@ -290,15 +227,7 @@ endif()
+@@ -293,15 +212,7 @@ endif()
  # - internal by default (CoinUtils),
  # - external if COINUTILS_INCLUDE_DIR_HINTS and a valid CoinUtils setup is found
  # ==============================================================================
 -if (NOT DEFINED COINUTILS_INCLUDE_DIR_HINTS)
```
Comment (JackBoosY) [inline ports/openmvg/fixcmake.patch]: Could you use `if (0) ... endif()` instead of delete them?

File: ports/openmvg/fixcmake.patch [hunk 6]
```diff
@@ -150,60 +158,60 @@ index 654bebd..ab7dd90 100644
...
 +find_package(CoinUtils REQUIRED)
  
  ## ==============================================================================
  ## Clp
-@@ -306,17 +235,7 @@ endif()
+@@ -309,17 +220,7 @@ endif()
  ## - internal by default (Clp),
  ## - external if CLP_INCLUDE_DIR_HINTS and a valid Clp setup is found
  ## ==============================================================================
 -if (NOT DEFINED CLP_INCLUDE_DIR_HINTS)
```
Comment (JackBoosY) [inline ports/openmvg/fixcmake.patch]: Could you use `if (0) ... endif()` instead of delete them?

File: ports/openmvg/fixcmake.patch [hunk 7]
```diff
@@ -150,60 +158,60 @@ index 654bebd..ab7dd90 100644
...
 +find_package(Clp REQUIRED)
  
  # ==============================================================================
  # Osi
-@@ -324,15 +243,7 @@ endif()
+@@ -327,15 +228,7 @@ endif()
  # - internal by default (Osi),
  # - external if OSI_INCLUDE_DIR_HINTS and a valid Osi setup is found
  # ==============================================================================
 -if (NOT DEFINED OSI_INCLUDE_DIR_HINTS)
```
Comment (JackBoosY) [inline ports/openmvg/fixcmake.patch]: Could you use `if (0) ... endif()` instead of delete them?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/netcdf-c/hdf5_3.patch (hunk 1 above)
Comment (Neumann-A) [reply ports/netcdf-c/hdf5_3.patch]: first check for TARGET hdf5::hdf5-static if not available try linking/checking hdf5::hdf5-shared

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Please resolve these conflicts.

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**